api/v1beta1/cluster_types.go

```diff
@@ -234,13 +234,19 @@ type MachineHealthCheckTopology struct {
 	MachineHealthCheckClass `json:",inline"`
 }
 
-// ClusterVariable can be used to customize the Cluster through
-// patches. It must comply to the corresponding
-// ClusterClassVariable defined in the ClusterClass.
+// ClusterVariable can be used to customize the Cluster through patches. Each ClusterVariable is associated with a
+// Variable definition in the ClusterClass `status` variables.
 type ClusterVariable struct {
 	// Name of the variable.
 	Name string `json:"name"`
 
+	// DefinitionFrom specifies where the definition of this Variable is from. DefinitionFrom is `inline` when the
+	// definition is from the ClusterClass `.spec.variables` or the name of a patch defined in the ClusterClass
+	// `.spec.patches` where the patch is external and provides external variables.
+	// This field is mandatory if the variable has `DefinitionsConflict: true` in ClusterClass `status.variables[]`
+	// +optional
+	DefinitionFrom *string `json:"definitionFrom"`
```

Comment:
Let's add omitempty

I think we don't have to make it a pointer given that unset and empty have the same semantic